File: android/src/main/kotlin/com/cloudwebrtc/webrtc/model/MediaType.kt
package com.cloudwebrtc.webrtc.model

import org.webrtc.MediaStreamTrack.MediaType as WMediaType;

enum class MediaType(val value: Int) {
    AUDIO(0),
    VIDEO(1);

    companion object {
        fun fromInt(value: Int) = values().first { it.value == value }
    }

    fun intoWebRtc(): WMediaType {
        return when (this) {
            AUDIO -> WMediaType.MEDIA_TYPE_AUDIO
            VIDEO -> WMediaType.MEDIA_TYPE_VIDEO
        }
    }
}

File: ios/Classes/model/MediaType.swift
import WebRTC

/// `RTCRtpMediaType` representation.
enum MediaType: Int {
  case audio
  case video

  /**
    Converts this `MediaType` into an `RTCRTPMediaType`.

    - Returns: `RTCRtpMediaType` based on this `MediaType`.
   */
  func intoWebRtc() -> RTCRtpMediaType {
    switch self {
    case .audio:
      return RTCRtpMediaType.audio
    case .video:
      return RTCRtpMediaType.video
    }
  }
}
